### PR TITLE
mempool: Stake-related readability improvements.

### PR DESCRIPTION
**Pull request by davecgh**
This does some minor cleanup of the code in the `maybeAcceptTransaction` function to use some local bools with more easily readable names for dealing with stake transactions.

It also improves the readability of the double spending exceptions section and avoids some additional if statements.

### CODE REVIEW

File: mempool/mempool.go [hunk 1]
```diff
@@ -902,60 +905,58 @@ func (mp *TxPool) maybeAcceptTransaction(tx *dcrutil.Tx, isNew, rateLimit, allow
...
+		for _, mpTx := range mp.pool {
+			if mpTx.Type == stake.TxTypeSSGen {
+				if mpTx.Tx.MsgTx().TxIn[1].PreviousOutPoint ==
+					msgTx.TxIn[1].PreviousOutPoint {
+					ssGenAlreadyFound++
 				}
-				if ssGenAlreadyFound > maxSSGensDoubleSpends {
+			}
+			if ssGenAlreadyFound > maxSSGensDoubleSpends {
+				str := fmt.Sprintf("transaction %v in the pool "+
```
Comment (dnldd) [inline mempool/mempool.go]: The error messages use`sstx`, `ssgen` and `ssrtx` which we are moving away from. Consider using `ticket`, `vote` and `revocation` instead.

File: mempool/mempool.go [hunk 2]
```diff
@@ -902,60 +905,58 @@ func (mp *TxPool) maybeAcceptTransaction(tx *dcrutil.Tx, isNew, rateLimit, allow
...
+	} else if isRevocation {
+		for _, mpTx := range mp.pool {
+			if mpTx.Type == stake.TxTypeSSRtx {
+				if mpTx.Tx.MsgTx().TxIn[0].PreviousOutPoint ==
+					msgTx.TxIn[0].PreviousOutPoint {
 					str := fmt.Sprintf("transaction %v in the pool "+
-						"with more than %v ssgens",
-						msgTx.TxIn[1].PreviousOutPoint,
-						maxSSGensDoubleSpends)
+						" as a ssrtx. Only one ssrtx allowed.",
```
Comment (dnldd) [inline mempool/mempool.go]: I think this error message could be more explicit about encountering multiple revocations referencing the same output.

File: mempool/mempool.go (hunk 1 above)
Comment (davecgh) [reply mempool/mempool.go]: Good point.  I didn't mess with cleaning up the errors as I was focused on the variables for upcoming PRs, but it makes sense to do so as a part of this PR.

File: mempool/mempool.go (hunk 2 above)
Comment (davecgh) [reply mempool/mempool.go]: Done.

**Pull request merged**